src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -2643,7 +2643,14 @@ static switch_status_t recog_channel_disable_all_grammars(speech_channel_t *scha
 {
 	switch_status_t status = SWITCH_STATUS_SUCCESS;
 
-	recognizer_data_t *r = (recognizer_data_t *) schannel->data;
+	recognizer_data_t *r;
```

Comment:
Init r to null otherwise r will be set to a garbage pointer. If schannel is deleted or set to null then accessing channel_destroyed is undefined behavior, it may crash, may return the old result or contain garbage value.

Reply:
Missed that. Thanks